packages/shared/src/components/buttons/BookmarkButton.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement, MouseEvent, ReactNode } from 'react';
+import { Post } from '../../graphql/posts';
+import ContextMenu from '../fields/ContextMenu';
+import { QuaternaryButton, QuaternaryButtonProps } from './QuaternaryButton';
+import { BookmarkIcon } from '../icons';
+import { SimpleTooltip } from '../tooltips';
+import { BookmarkReminderIcon } from '../icons/Bookmark/Reminder';
+import useContextMenu from '../../hooks/useContextMenu';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { LazyModal } from '../modals/common/types';
+import { useBookmarkReminder } from '../../hooks/notifications';
+import { ButtonColor, ButtonVariant } from './Button';
+
+interface BookmarkButtonProps {
+  buttonProps?: QuaternaryButtonProps<'button'>;
+  contextMenuId?: string;
+  post: Post;
+  children?: ReactNode;
+}
+
+export function BookmarkButton({
+  contextMenuId = 'bookmark-action',
+  buttonProps = {},
+  post,
+  children,
+}: BookmarkButtonProps): ReactElement {
+  const finalId = `${contextMenuId}-${post.id}`;
+  const hasReminder = !!post.bookmark?.remindAt;
+  const { openModal } = useLazyModal();
+  const { onRemoveReminder } = useBookmarkReminder();
+  const { onMenuClick } = useContextMenu({ id: finalId });
+  const Icon = hasReminder ? BookmarkReminderIcon : BookmarkIcon;
+  const onClick = (e: MouseEvent<HTMLButtonElement>) => {
+    if (hasReminder) {
+      return onMenuClick(e);
+    }
+
+    return buttonProps.onClick?.(e);
+  };
+
+  return (
+    <>
+      <SimpleTooltip content={post.bookmarked ? 'Remove bookmark' : 'Bookmark'}>
+        <QuaternaryButton
```

Comment:
Are you sure this is right though?
I mean the current actionButtons render as:

```
<Button
            id={`post-${post.id}-bookmark-btn`}
            className="pointer-events-auto ml-2"
            icon={<BookmarkIcon secondary={post.bookmarked} />}
            onClick={() => onBookmarkClick(post)}
            color={ButtonColor.Bun}
            pressed={post.bookmarked}
            variant={ButtonVariant.Float}
          />
```

Reply:
Let me check where this is.

Reply:
Check the difference between the buttons. Should practically be the same, the point of the Quaternary button is to have a wrapper so that when you add a label, the label will not have the same background as the button. Since this is for the list feed, we don't show any label from the actions there in any format. We should be good on that end.

Reply:
I think we should be able to merge here, and if we really want to avoid the extra wrapper, I think we can work on it if the experiment wins. Thoughts?